### ISSUE TITLE
Load multiple source/target file?

### ISSUE THREAD

**Issue by YuChunLOL** [OpenNMT/OpenNMT-py#482]
Hello, I have a question about here:
https://github.com/OpenNMT/OpenNMT-py/blob/master/train.py#L216

I noticed that the new function `load_dataset()` in `train.py` seems to support multiple files/datasets?
but it seems that the current `preprocess.py`/`opts.py` did not support this feature yet (It only return [dataset] here: https://github.com/OpenNMT/OpenNMT-py/blob/master/preprocess.py#L171)

Thanks.

**Comment by JianyuZhan** [comment]
No, only support one files.  Multiple datasets actually are internal implementation used to tackle huge file, see https://github.com/OpenNMT/OpenNMT-py/blob/master/preprocess.py#L144

**Comment by srush** [comment]
@YuChunLOL  Can I ask the application you are interested in?

**Comment by YuChunLOL** [comment]
@srush My task is grammatical error correction, where my test set has 4 references for each source sentence. See test data here: https://github.com/keisks/jfleg/tree/master/test

**Comment by YuChunLOL** [comment]
Currently, I hacked this by concatenating 4 reference file in a single reference file, and also concatenating 4 times of source file to a single source file.

**Comment by srush** [comment]
Cool, that seems like a feasible solution for now. 

In theory, you might want to model the fact that there are multiple different corrections. We don't currently support that, but would be happy to hear other ideas.

**Comment by howardyclo** [comment]
@srush  Thanks for your advice! I'll survey that how to model one-to-many (or many-to-one) mapping Currently I have no idea, so for training, the src/tgt are one-to-one, but for validation and testing, there're one-to-many. It would be great if professor can suggest me some directions :-D !

**Comment by srush** [comment]
Oh I see. In that case, if it is just validation and testing, your method is fine. 

The problem is that during decoding, you may need to retune towards your final metric. See my students paper on the topic https://arxiv.org/pdf/1707.09067.pdf

**Comment by laotao** [comment]
I'm facing the same problem and find this paper: https://www.aclweb.org/anthology/P18-2053
